### ISSUE TITLE
[avrogen] Avro int32s are mapped to Go ints

### ISSUE THREAD

**Issue by sk-** [hamba/avro#395]
First of all thanks or this library. We are assessing it as a replacement of gogen-avro, which is on a [permanent break](https://github.com/actgardner/gogen-avro), and we are quite pleased with the results so far.

As shown in [Avro's specification](https://avro.apache.org/docs/1.11.1/specification/#primitive-types), an `int` is an signed integer using 32 bits, which correspond to the Go type `int32`. However, the current [generator specifies](https://github.com/hamba/avro/blob/0e4c8f96537226fa1eebba258fb2691a87b41fc5/gen/gen.go#L50) them as `int`s, which use 32 or 64 bits depending on the architecture.

This has the problem, that when constructing an Avro message one could be silently losing precision. It would be better if it could be explicit that the underlying type has 32 bits.

**Comment by nrwiersma** [comment]
Fair point. I think it should allow for explicit `int32` use, maybe via a flag, as 1) it would be backwards breaking otherwise and 2) many people, myself included, hate having to always cast when largely dealing with `int`s in code.

A PR to add this support would be welcome, otherwise I will try get to it in the near future.

**Comment by nrwiersma** [comment]
Fixed by #396
